### PR TITLE
Dissallow transactions to be sent to 0x000...

### LINKED ISSUE #1544: dont allow sending transactions to 0x0000000000000000000000000000000000000000
dont allow sending transactions to 0x0000000000000000000000000000000000000000 be noisy and show an error on send screen and conf-tx

### PR DESCRIPTION
**Pull request by frankiebee**
fixes #1544 
screen shot 2017-06-05 at 11 53 26 am
screen shot 2017-06-05 at 11 53 52 am
screen shot 2017-06-05 at 11 53 57 am

### CODE REVIEW

File: ui/app/send.js [hunk 1]
```diff
@@ -262,7 +262,7 @@ SendTransactionScreen.prototype.onSubmit = function () {
     return this.props.dispatch(actions.displayWarning(message))
   }
 
-  if ((!util.isValidAddress(recipient) && !txData) || (!recipient && !txData)) {
+  if ((!util.isValidAddress(recipient) && !txData) || (!recipient && !txData) || (recipient === '0x0000000000000000000000000000000000000000')) {
```
Comment (kumavis) [inline ui/app/send.js]: maybe move '0x0000000000000000000000000000000000000000' check into `isValidAddress`

File: ui/app/components/pending-tx.js [hunk 2]
```diff
@@ -44,6 +44,9 @@ PendingTx.prototype.render = function () {
   const account = props.accounts[address]
   const balance = account ? account.balance : '0x0'
 
+  // recipient check
+  const isValidAddress = !(txParams.to === '0x0000000000000000000000000000000000000000')
```
Comment (kumavis) [inline ui/app/components/pending-tx.js]: should we use `util.isValidAddress(txParams.to)` here?

File: ui/app/components/pending-tx.js [hunk 3]
```diff
@@ -261,6 +264,15 @@ PendingTx.prototype.render = function () {
           }, 'Transaction Error. Exception thrown in contract code.')
         : null,
 
+        !isValidAddress ?
+          h('.error', {
+            style: {
+              marginLeft: 50,
+              fontSize: '0.9em',
+            },
+          }, 'Recipient address is invalid sending this transaction will result in a loss of ETH.')
```
Comment (kumavis) [inline ui/app/components/pending-tx.js]: > Sending this transaction will result in a loss of ETH.

do we show a warning or do we disallow approval?
if we dont allow this, we dont need to explain the consequences.
> Recipient address is invalid.
should suffice

File: ui/app/components/pending-tx.js [hunk 4]
```diff
@@ -298,7 +310,7 @@ PendingTx.prototype.render = function () {
             type: 'submit',
             value: 'ACCEPT',
             style: { marginLeft: '10px' },
-            disabled: insufficientBalance || !this.state.valid,
+            disabled: insufficientBalance || !this.state.valid || !isValidAddress,
```
Comment (kumavis) [inline ui/app/components/pending-tx.js]: `this.state.valid` and `isValidAddress`
are we being redundant here? maybe not but just wanted to check

Comment (Zanibas) [inline ui/app/components/pending-tx.js]: One of these may be checking if the gas values are correct, while the other one is checking if the address is valid. We may be able to merge the logic for checking isValidAddress into a `setState`

Comment (Zanibas) [inline ui/app/components/pending-tx.js]: I think this is fine though, since we need the separate check anyways to display a conditional error message.

Comment (kumavis) [inline ui/app/components/pending-tx.js]: im suspicious of `setState` being async and leaving a window where validation hasnt caught up

but not sure how possible that is

**Comment by kumavis** [comment]
if you change the address from a valid one to an invalid one, it will use the old valid value

reproduce:
- write `dinodan.eth`
- wait for resolve
- change to `fenbushi.eth`
- click send

**Review by kumavis — CHANGES_REQUESTED** [review]
need to address the issue where it uses a previously good name even if a bad name is currently set

**Review by 2-am-zzz — DISMISSED** [review]
Related to network switching, if I start the send screen in Ropsten, switch to Main, and then type a valid address, the tooltip gets an RPC error. Also works the reverse. Not really related to this persay, but found in QA.

**Review by kumavis — APPROVED** [review]

**Comment by kumavis** [comment]
i diggity dig it

**Pull request merged**